DependencyInjection/SonataUserExtension.php

```diff
@@ -88,12 +88,12 @@ public function load(array $configs, ContainerBuilder $container)
         $profileFormDefinition = $container->getDefinition('sonata.user.profile.form');
         $registrationFormDefinition = $container->getDefinition('sonata.user.registration.form');
         if (method_exists($profileFormDefinition, 'setFactory')) {
-            $profileFormDefinition->setFactory(array('form.factory', 'createNamed'));
-            $registrationFormDefinition->setFactory(array('form.factory', 'createNamed'));
+            $profileFormDefinition->setFactory(array(new Reference('form.factory'), 'createNamed'));
+            $registrationFormDefinition->setFactory(array(new Reference('form.factory'), 'createNamed'));
         } else {
-            $profileFormDefinition->setFactoryClass('form.factory');
+            $profileFormDefinition->setFactoryClass(new Reference('form.factory'));
```

Comment:
AFAIK both ways works, on MediaAdmin there is the other way: https://github.com/sonata-project/SonataMediaBundle/blob/3.x/DependencyInjection/SonataMediaExtension.php#L49

On this bundle we used a reference on previous version: https://github.com/sonata-project/SonataUserBundle/blob/3.2.1/Resources/config/form.xml#L5

That's why I left it with a Reference... to preserve original code

If you want me to change it, I can do it.